packages/flutter_test/lib/src/widget_tester.dart

```diff
@@ -438,10 +440,36 @@ class WidgetTester extends WidgetController implements HitTestDispatcher, Ticker
     return TestAsyncUtils.guard<void>(() {
       binding.attachRootWidget(widget);
       binding.scheduleFrame();
+      if (_firstPumpWidget) {
+        _firstPumpWidget = false;
+        // Emulate what `RenderObjectToWidgetAdapter.attachToRenderTree` does
+        // when it builds a new tree from scratch.
+        binding.buildOwner.buildScope(binding.renderViewElement);
+      }
       return binding.pump(duration, phase);
     });
   }
 
+  /// This is the old implementation of [pumpWidget] that was replaced in
+  /// https://github.com/flutter/flutter/pull/48922.
+  ///
+  /// Differences are explained in the tests in `test/widget_tester_test.dart`.
+  ///
+  /// The only reason we still have this is because a few material text field
+  /// tests are failing with the new implementation of [pumpWidget]. See
+  /// https://github.com/flutter/flutter/issues/49077.
+  @Deprecated(
+    'Use WidgetTester.pumpWidget() instead. '
+    'This feature was deprecated after v1.14.1.'
+  )
+  Future<void> pumpWidgetLegacy(Widget widget) {
+    return TestAsyncUtils.guard<void>(() {
+      binding.attachRootWidget(widget);
+      binding.scheduleFrame();
+      return binding.pump();
+    });
+  }
```

Comment:
please fix the tests rather than leaving them like this, otherwise we're just accumulating technical debt.